app/code/Magento/Checkout/view/frontend/web/js/region-updater.js

```diff
@@ -42,6 +48,10 @@ define([
                 this.currentRegionOption = $(e.target).val();
             }, this));
 
+            $(this.options.regionInputId).on('change', $.proxy(function (e) {
+                this.currentRegionValue[this.element.val()] = $(e.target).val();
+            }, this));
+
             $(this.options.regionInputId).on('focusout', $.proxy(function () {
```

Comment:
Only one remark from my side: `$(this.options.regionInputId)` is now called twice, could you either introduce `regionInput` variable at the top of the function and reuse it or chain the `.on` calls together? I don't lean towards any of the solutions, decide which looks cleaner yourself.